libraries/AP_Compass/AP_Compass.h

```diff
@@ -143,6 +143,9 @@ friend class AP_Compass_Backend;
     bool compass_cal_requires_reboot() const { return _cal_complete_requires_reboot; }
     bool is_calibrating() const;
 
+    // indicate which bit in LOG_BITMASK indicates we should log compass readings
+    void set_log_bit(uint32_t log_bit) { _log_bit = log_bit; }
```

Comment:
Why do this as a separate function instead of passing in init or the constructor like other libraries do?